src/v2i-hub/RSUHealthMonitorPlugin/src/RSUConfigurationList.cpp

```diff
@@ -0,0 +1,122 @@
+
+#include "RSUConfigurationList.h"
+
+namespace RSUHealthMonitor
+{
+    Json::Value RSUConfigurationList::parseJson(const std::string &rsuConfigsStr) const
+    {
+        JSONCPP_STRING err;
+        Json::Value root;
+        auto length = static_cast<int>(rsuConfigsStr.length());
+        Json::CharReaderBuilder builder;
+        std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+        if (!reader->parse(rsuConfigsStr.c_str(), rsuConfigsStr.c_str() + length, &root, &err))
+        {
+            std::stringstream ss;
+            ss << "Parse RSUs raw string error: " << err;
+            throw RSUConfigurationException(ss.str().c_str());
+        }
+        return root;
+    }
+
+    void RSUConfigurationList::parseRSUs(const std::string &rsuConfigsStr)
+    {
+        auto json = parseJson(rsuConfigsStr);
+        std::vector<RSUConfiguration> tempConfigs;
+        RSUConfiguration config;
+        auto rsuArray = json[RSUSKey];
+        if (!rsuArray.isArray())
+        {
+            throw RSUConfigurationException("RSUS is not an array.");
+        }
+        for (auto i = 0; i != rsuArray.size(); i++)
+        {
+            if (rsuArray[i].isMember(RSUIpKey))
+            {
+                config.rsuIp = rsuArray[i][RSUIpKey].asString();
+            }
+            else
+            {
+                throw RSUConfigurationException("RSU IP does not exist.");
```

Comment:
Shouldn't each RSU have an ID or name as well. Then instead of `RSU IP does not exist` we could have a more descriptive exception. `IP is missing for RSU <name>`. Otherwise when this exception is thrown and you have 4 RSU configs, only one has a miss spelling of the RSUIPKey then it maybe hard to find. Just a thought.

Reply:
yeah, I agree.